Update fetch-and-compile readme with fetchAndCompileMultiple and getSupportedNetworks

haltman-at opened the pull request:
Thanks to @lsqproduction for noticing we'd forgotten to update this.  Hoping these usage examples are sufficient?

lsqproduction reviewed (commented):
Looks good. Thank you for adding it.

cds-amal reviewed (commented):
I created a new TS project to verify this snippet and ran into issues simply importing `@Truffle/fetch-and-compile` and trying to compile.

``` shell
$ tsc   
node_modules/@truffle/compile-common/dist/src/types.d.ts:1:47 - error TS2307: Cannot find module '@truffle/contract-schema/spec' or its corresponding type declarations.

1 import type { Abi, ImmutableReferences } from "@truffle/contract-schema/spec";
         
```

<details><summary>tsconfig.json</summary>

``` json
{
  "compilerOptions": {
    "target": "es2016",
    "module": "commonjs",
    "esModuleInterop": true,
    "strict": true,
    "outDir": "dist",
    "sourceMap": true
  }
}
```

</details>

<details><summary>package.json</summary>
```
{
  "name": "fetch-and-compile",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@truffle/fetch-and-compile": "^0.4.0"
  }
}
```
</details>

haltman-at commented on packages/fetch-and-compile/README.md:
````diff
@@ -65,5 +65,56 @@ async function decode(address: string) {
 decode("0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e");
 ```
 
+If you want to fetch and compile multiple contracts from the same network, you can use `fetchAndCompileMultiple`:
````
OK, I'll go back and do that.

haltman-at commented:
OK, I added headers as @gnidan suggested, and added a note about `skipLibCheck` as @cds-amal suggested elsewhere.  Note that the `skipLibCheck` thing really applies to a number of our packages!  (Amal do you maybe want to put up an issue for that...?)  Seeing as ultimately the problem comes from contract-schema, which a number of our packages depend on.

cds-amal commented on packages/fetch-and-compile/README.md:
```diff
@@ -2,7 +2,11 @@
 This is used to obtain external verified sourced and compile them.
 
-### Usage
+Note: If you import this into your TS project, you may need to set `skipLibCheck` in your tsconfig due to an indirect dependency on @truffle/contract-schema.
```
**Set** is ambiguous here, perhaps **enable**, or explicitly "set skipLibCheck to true"

cds-amal reviewed (approved):
Looks good! Thanks for the help getting this tested
